### ISSUE TITLE
'alternates' support doesn't conform to git's repo layout specification

### ISSUE THREAD

**Issue by leith-bartrich** [git-lfs/git-lfs#3115]
Merge request #1007 adds support for a local reference for fetch operations.  Which is good.  I love it.  I would love to see that support expanded actually.

However, upon looking into it, I found it doesn't conform to git's specification.

Specifically, the "alternates" file is supposed to take multiple alternate paths on separate lines.  The code that was merged in to git-lfs, assumes only a single line.  An "alternates" file with multiple lines shows an empty "LocalReferenceDir=" when calling "git lfs env".  Which makes sense given the way it was implemented.  But this behavior does rail against the git repository layout doc.  https://git-scm.com/docs/gitrepository-layout

The easiest semi-fix would be to just take the first line, rather than the whole file.  But really, the support should be updated to extract and cascade down a list of alternates and use all the valid ones in order, on each object.  Ideally, in the same order git's internal code implements.

Further (And maybe this is  a separate issue) I'd think this is a great place to intelligently support local clones, pulls and fetches; by checking the remote's object path and adding it to the list of LocalReferenceDir(s) if it is in fact fetching from a local path.  Then, adding a flag to disable this behavior, or alternately a flag to enable the behavior.

**Comment by leith-bartrich** [comment]
I've done a local checkout of git lfs and taught myself enough go to hack up the fix and change referenced above.

Local testing on my linux successfully executes local clones via a fixed version of the 'alternates' mechanism, where it's using multiple alternates, rather than just one. 

Is this something you'd want to see a pull request for?  Again, I'm no go expert.

**Comment by ttaylorr** [comment]
@leith-bartrich Thank you so much for digging into this issue! I thought that I had already commented here, but I have already opened up #3116, and #3117 which I think (taken together) should fix the issue.

If you have time, would you mind trying them out for yourself? If they don't work as expected, please feel more than free to open up a pull request, or leave a comment on mine.

I'm sorry that I forgot to leave a note here that I added support for this, but I appreciate you taking the time out to learn some Go and work on Git LFS nonetheless.

**Comment by leith-bartrich** [comment]
I pulled down #3117 and found it didn't build.  Looking deeper, I get the same build errors your own CI system is flagging.  That having been said, I took a look at what you did and your code is very similar to what I did.  I'm confident it's the right set of changes.  And the added tests are much appreciated as well.  I assume we should close this issue and continue testing of the pull request.  I had an additional feature I added.  But I'll file a new issue or pull request once #3117 settles and merges in.  The environment variable support you added is also greatly appreciated.  Thank you.  I'll leave it to you to close the issue just in-case you want to keep it open.
